### PR TITLE
[ONNX] Fix for dynamic scale_factor export

### PR DESCRIPTION
**Pull request by neginraoof**
Script scale_factor computation to enable dynamic input shape export.

[Tests in this PR will pass after https://github.com/pytorch/pytorch/pull/35744 is merged, so this PR needs to wait]

### CODE REVIEW

**Comment by neginraoof** [comment]
@fmassa I think the interpolate PR is close to merge. Please review this one, and I'll trigger it again after the other PR went in. Thanks a lot.

File: torchvision/models/detection/transform.py [hunk 1]
```diff
@@ -76,17 +91,12 @@ def torch_choice(self, l):
...
-        max_size = float(torch.max(im_shape))
         if self.training:
             size = float(self.torch_choice(self.min_size))
         else:
             # FIXME assume for now that testing uses the largest scale
             size = float(self.min_size[-1])
-        scale_factor = size / min_size
-        if max_size * scale_factor > self.max_size:
-            scale_factor = self.max_size / max_size
+        scale_factor = compute_scale_factor(image, size, float(self.max_size))
```
Comment (fmassa) [inline torchvision/models/detection/transform.py:99]: This will probably not work for torchscript because you marked it as  as `torch.jit.unused`.

What I think you'll need to do is instead  to have two functions, one for torchscript and one for the ONNX codepath, which gets activated when the code is tracing.

Comment (neginraoof) [reply torchvision/models/detection/transform.py:99]: I had to mark it as unused to be able to do "from torch.onnx import operators" in the method.
With torch.jit.script, this was not possible. But I can take the import out and mark is as torch.jit.script. Or create a branch for ONNX

Comment (fmassa) [reply torchvision/models/detection/transform.py:99]: Please create a branch for ONNX, otherwise this will add a hard-dependency on ONNX in torchvision.

I'm thinking about something like
```python
if torchvision._is_tracing():
    scale_factor = _compute_scale_factor_onnx(...)
else:
    scale_factor = _compute_scale_factor(...)
```

File: torchvision/models/detection/transform.py [hunk 2]
```diff
@@ -76,17 +89,17 @@ def torch_choice(self, l):
...
             size = float(self.torch_choice(self.min_size))
         else:
             # FIXME assume for now that testing uses the largest scale
             size = float(self.min_size[-1])
-        scale_factor = size / min_size
-        if max_size * scale_factor > self.max_size:
-            scale_factor = self.max_size / max_size
+        if torchvision._is_tracing():
+            from torch.onnx import operators
+            im_shape = operators.shape_as_tensor(image)[-2:]
```
Comment (neginraoof) [inline torchvision/models/detection/transform.py:99]: @fmassa Please review this update. Since most of the code is similar in two paths, I updated the branch as such.

File: torchvision/models/detection/transform.py [hunk 3]
```diff
@@ -10,6 +10,37 @@
 from .roi_heads import paste_masks_in_image
 
 
+@torch.jit.unused
```
Comment (neginraoof) [inline torchvision/models/detection/transform.py:13]: cc @fmassa 
I implemented this separate code path to avoid constant sale_factor. Issue is describe in: https://github.com/pytorch/pytorch/issues
Since 'scale_factor' was used to resize the image as well as target['masks'], I had to factor out this part of the code together.

**Comment by neginraoof** [comment]
Thank you @fmassa
